features/steps/steps.go

```diff
@@ -1,13 +1,263 @@
+	url := fmt.Sprintf(
+		"/datasets/%s/editions/%s/versions/%s/dimensions/%s/options?%s",
+		datasetID,
+		edition,
+		version,
+		dimension,
+		params,
+	)
+
+	c.DatasetAPI.NewHandler().
+		Get(url).
+		Reply(http.StatusOK).
+		BodyString(o.Content)
+
+	return nil
+}
+// theFollowingDimensionsAreAvailable generates a mocked response for dataset API
+// GET /datasets/{dataset_id}/editions/{edition}/versions/{version}/dimensions
+func (c *Component) theFollowingDimensionsAreAvailable(datasetID, edition, version string, d *godog.DocString) error {
+	url := fmt.Sprintf(
+		"/datasets/%s/editions/%s/versions/%s/dimensions",
+		datasetID,
+		edition,
+		version,
+	)
+
+	c.DatasetAPI.NewHandler().
+		Get(url).
+		Reply(http.StatusOK).
+		BodyString(d.Content)
+
+	return nil
+}
+
+// theFollowingVersionDocumentIsAvailable generates a mocked response for dataset API
+// GET /datasets/{dataset_id}/editions/{edition}/versions/{version}
+func (c *Component) theFollowingVersionDocumentIsAvailable(datasetID, edition, version string, v *godog.DocString) error {
+	url := fmt.Sprintf(
+		"/datasets/%s/editions/%s/versions/%s",
+		datasetID,
+		edition,
+		version,
+	)
+
+	c.DatasetAPI.NewHandler().
+		Get(url).
+		Reply(http.StatusOK).
+		BodyString(v.Content)
+
+	return nil
 }
 
-func (c *Component) iShouldReceiveAHelloworldResponse() error {
+// theFollowingVersionIsUpdated generates a mocked response for dataset API
+// PUT /instances/{id} with the provided instance response
+func (c *Component) theFollowingVersionWillBeUpdated(datasetID, edition, version string, v *godog.DocString) error {
+	url := fmt.Sprintf(
+		"/datasets/%s/editions/%s/versions/%s",
+		datasetID,
+		edition,
+		version,
+	)
+
+	c.DatasetAPI.NewHandler().
+		Put(url).
+		AssertCustom(newPutVersionAssertor([]byte(v.Content))).
+		Reply(http.StatusOK)
+
+	return nil
+}
+
+func (c *Component) thisCantabularMetadataExportEventIsConsumed(input *godog.DocString) error {
+	ctx := context.Background()
+
+	// testing kafka message that will be produced
+	var testEvent event.CSVCreated
+	if err := json.Unmarshal([]byte(input.Content), &testEvent); err != nil {
+		return fmt.Errorf("error unmarshaling input to event: %w body: %s", err, input.Content)
+	}
+
+	log.Info(ctx, "event to marshal: ", log.Data{
+		"event": testEvent,
+	})
+
+	// marshal and send message
+	b, err := schema.CSVCreated.Marshal(testEvent)
+	if err != nil {
+		return fmt.Errorf("failed to marshal event from schema: %w", err)
+	}
+
+	log.Info(ctx, "marshalled event: ", log.Data{
+		"event": b,
+	})
+
+	c.producer.Channels().Output <- b
+	return nil
+}
+
+func (c *Component) theFollowingFileCanBeSeenInMinio(fileName string, bucket string) error {
+	ctx := context.Background()
+
+	var b []byte
+	f := aws.NewWriteAtBuffer(b)
+
+	// probe bucket with backoff to give time for event to be processed
+	retries := 5
```

Comment:
I'd suggest using constants, similar to other repositories, for example:
https://github.com/ONSdigital/dp-cantabular-csv-exporter/blob/be9bb7dee2bbe104536bac61befe7d15c24cd297/features/steps/steps.go#L251